src/main/java/com/avispl/symphony/dal/infrastructure/management/samsung/smartthings/SamsungSmartThingsAggregatorCommunicator.java

```diff
@@ -0,0 +1,3177 @@
+			} else {
+				logger.error("Hub health info is empty, Please go to https://graph-ap02-apnortheast2.api.smartthings.com/ to check if the hub are removed from Smartthings");
+			}
+		} catch (Exception e) {
+			if (retryOnError) {
+				retrieveHubHealth(stats, hubId, false);
+			}
+			logger.error("Error while retrieve hub info: " + e.getMessage(), e);
+		}
+	}
+
+	/**
+	 * This method is used to retrieve locations info by send GET request to "https://api.smartthings.com/v1/locations"
+	 *
+	 * @param retryOnError retry if any error occurs
+	 * @throws ResourceNotReachableException If any error occurs
+	 */
+	public void retrieveLocations(boolean retryOnError) {
+		try {
+			String request = SmartThingsURL.LOCATIONS;
+
+			LocationWrapper locationWrapper = doGet(request, LocationWrapper.class);
+
+			if (locationWrapper != null && locationWrapper.getLocations() != null) {
+				cachedLocations = locationWrapper.getLocations();
+				locationIdFiltered = findLocationByName(locationFilter).getLocationId();
+			} else {
+				logger.error("Locations is empty, Please go to https://graph-ap02-apnortheast2.api.smartthings.com/ to check if the locations are removed from Smartthings");
+			}
+		} catch (Exception e) {
+			if (retryOnError) {
+				retrieveLocations(false);
+			}
+			logger.error("Error while retrieve locations info: " + e.getMessage(), e);
+
+		}
+	}
+
+	/**
+	 * This method is used to populate locations management group
+	 *
+	 * @param stats store all statistics
+	 * @param advancedControllableProperties store all controllable properties
+	 */
+	public void populateLocationsManagement(Map<String, String> stats, List<AdvancedControllableProperty> advancedControllableProperties) {
+		for (int locationIndex = 0; locationIndex < cachedLocations.size(); locationIndex++) {
+			addAdvanceControlProperties(advancedControllableProperties, createText(stats, AggregatorGroupControllingMetric.LOCATION_MANAGEMENT.getName()
+							+ LocationManagementMetric.LOCATION.getName() + formatOrderNumber(locationIndex, Arrays.asList(cachedLocations.toArray())),
+					getDefaultValueForNullData(cachedLocations.get(locationIndex).getName(), SmartThingsConstant.NONE)));
+		}
+	}
+
+	/**
+	 * This method is used to populate current location
+	 *
+	 * @param stats store all statistics
+	 */
+	public void populateCurrentLocation(Map<String, String> stats) {
+		String locationName = cachedLocations.stream().filter(l -> l.getLocationId().equals(locationIdFiltered)).map(Location::getName).findFirst()
+				.orElse(cachedLocations.get(0).getName());
+		stats.put(HubInfoMetric.CURRENT_LOCATION.getName(), locationName);
+	}
+
+	/**
+	 * This method is used to retrieve rooms info by send GET request to "https://api.smartthings.com/v1/locations/{locationId}/rooms"
+	 *
+	 * @param retryOnError retry if any error occurs
+	 * @throws ResourceNotReachableException When getting the empty response
```

Comment:
it doesn't seem like this method really throws anything

Reply:
I fixed it